diff --git src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD
index 0a91766e2e..ce0a53bbc0 100644
--- src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD
+++ src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD
@@ -28,3 +28,20 @@ proto_library(
         "//src/main/protobuf:invocation_policy_proto",
     ],
 )
+
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
+
+go_proto_library(
+    name = "build_event_stream_go_proto",
+    importpath = "github.com/bazelbuild/bazel/src/main/java/com/google/devtools/build/lib/buildeventstream/proto",
+    proto = ":build_event_stream_proto",
+    deps = ["//src/main/protobuf:go_default_library"],
+)
+
+go_library(
+    name = "go_default_library",
+    embed = [":build_event_stream_go_proto"],
+    importpath = "github.com/bazelbuild/bazel/src/main/java/com/google/devtools/build/lib/buildeventstream/proto",
+    visibility = ["//visibility:public"],
+)
diff --git src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto
index 79e11447b9..c7bf9c4765 100644
--- src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto
+++ src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto
@@ -21,6 +21,7 @@ import "src/main/protobuf/invocation_policy.proto";
 
 option java_package = "com.google.devtools.build.lib.buildeventstream";
 option java_outer_classname = "BuildEventStreamProtos";
+option go_package = "github.com/bazelbuild/bazel/src/main/java/com/google/devtools/build/lib/buildeventstream/proto";
 
 // Identifier for a build event. It is deliberately structured to also provide
 // information about which build target etc the event is related to.
diff --git src/main/protobuf/BUILD src/main/protobuf/BUILD
index 8056e19349..d5c6f8df70 100644
--- src/main/protobuf/BUILD
+++ src/main/protobuf/BUILD
@@ -204,3 +204,35 @@ filegroup(
         ":remote_execution_log_java_proto_srcs",
     ],
 )
+
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//proto:def.bzl", "go_proto_library")
+
+go_proto_library(
+    name = "command_line_go_proto",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    proto = ":command_line_proto",
+)
+
+go_proto_library(
+    name = "invocation_policy_go_proto",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    proto = ":invocation_policy_proto",
+)
+
+go_proto_library(
+    name = "option_filters_go_proto",
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    proto = ":option_filters_proto",
+)
+
+go_library(
+    name = "go_default_library",
+    embed = [
+        ":command_line_go_proto",
+        ":option_filters_go_proto",
+        ":invocation_policy_go_proto",
+    ],
+    importpath = "github.com/bazelbuild/bazel/src/main/protobuf",
+    visibility = ["//visibility:public"],
+)
diff --git src/main/protobuf/command_line.proto src/main/protobuf/command_line.proto
index d5fa6aceb8..07fac0a85a 100644
--- src/main/protobuf/command_line.proto
+++ src/main/protobuf/command_line.proto
@@ -17,6 +17,7 @@ package command_line;
 
 // option java_api_version = 2;
 option java_package = "com.google.devtools.build.lib.runtime.proto";
+option go_package = "github.com/bazelbuild/bazel/src/main/protobuf";
 
 import "src/main/protobuf/option_filters.proto";
 
diff --git src/main/protobuf/invocation_policy.proto src/main/protobuf/invocation_policy.proto
index 53572ac120..27f5889452 100644
--- src/main/protobuf/invocation_policy.proto
+++ src/main/protobuf/invocation_policy.proto
@@ -17,6 +17,7 @@ package blaze.invocation_policy;
 
 // option java_api_version = 2;
 option java_package = "com.google.devtools.build.lib.runtime.proto";
+option go_package = "github.com/bazelbuild/bazel/src/main/protobuf";
 
 // The --invocation_policy flag takes a base64-encoded binary-serialized or text
 // formatted InvocationPolicy message.
diff --git src/main/protobuf/option_filters.proto src/main/protobuf/option_filters.proto
index 782620d053..a6fc797136 100644
--- src/main/protobuf/option_filters.proto
+++ src/main/protobuf/option_filters.proto
@@ -16,6 +16,7 @@ package options;
 
 // option java_api_version = 2;
 option java_package = "com.google.devtools.common.options.proto";
+option go_package = "github.com/bazelbuild/bazel/src/main/protobuf";
 
 // IMPORTANT NOTE: These two enums must be kept in sync with their Java
 // equivalents in src/main/java/com/google/devtools/common/options.
